### PR TITLE
Make `Shop::getBaseURL` use secure mode by default

### LINKED ISSUE #28468: Shop::getBaseURL should return the good url sheme: http or https
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

By default $shop->getBaseURL() without the first argument return shop url in http only.

### Expected behavior

https is the standard now, getBaseURL() should return the good scheme by default and not http by default.

There are https calls of getBaseURL() without argument in Prestashop core and native modules (QuickAccess.php, HelperForm, ps_mainmenu, ps_imageslider, ...)

### Steps to reproduce

1. Call Context::getContext()->shop->getBaseURL(); in a shop in https in front
2. Display the URL

### PrestaShop version(s) where the bug happened

1.7, 8

### PHP version(s) where the bug happened

all

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Shop::getBaseURL return the good url sheme by default: http or https
| Type?             | improvement
| Category?         | FO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/28468
| Related PRs       | 
| How to test?      | See issue
| Possible impacts? | BC Break

BC break: the first argument auto_secure_mode is set to true by default now.

### CODE REVIEW

File: classes/shop/Shop.php [hunk 1]
```diff
@@ -521,7 +521,7 @@ public function getBaseURI()
      *
      * @return string|bool complete base url of current shop
      */
-    public function getBaseURL($auto_secure_mode = false, $add_base_uri = true)
+    public function getBaseURL($auto_secure_mode = true, $add_base_uri = true)
```
Comment (PululuK) [inline classes/shop/Shop.php:524]: Should we check if `PS_SSL_ENABLED` is enabled ?

Comment (jf-viguier) [reply classes/shop/Shop.php:524]: There is two ways to get url sheme in prestashop
- check PS_SSL_ENABLED 
- Tools::usingSecureMode : that check if https is activated in server envs

In Link::getBaseLink, http or https is checked against PS_SSL_ENABLED and PS_SSL_ENABLED_EVERYWHERE

This is not the point of my pr to change this two ways.

Comment (PululuK) [reply classes/shop/Shop.php:524]: so even if `PS_SSL_ENABLED` is disabled the `Shop::getBaseURL` will always use secure mode by default?

Comment (jf-viguier) [reply classes/shop/Shop.php:524]: No, Shop::getBaseURL will check if the context is in https or http by defaut.

**Comment by jf-viguier** [comment]
Can someone remove "Waiting for wording" label, it's not.

**Review by matks — APPROVED** [review]
OK for me

**Review by MeKeyCool — APPROVED** [review]
I assume that you checked all `getBaseURL` calls.
Most of them use default value (so `false`) and never set explicit `false` value.

Then, if it is ok to transform all `false` to true, what is the point to keep that argument ?
If BC is too brutal, maybe at least setting it to deprecated ?

What do you think ? :thinking:

**Comment by jf-viguier** [comment]
@MeKeyCool other getURL methods always check https, my pr homogenizes this behavior.
Please read me comment here :
https://github.com/PrestaShop/PrestaShop/pull/28469#discussion_r871371946

**Review by Progi1984 — APPROVED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @jf-viguier ,

Thanks for the PR.

I've tested it on develop branch.

With SSL activated options enabled :
Screenshot 2022-06-13 at 12 07 00

I have added {$urls|@var_dump} to my .tpl, I have this result : 
Screenshot 2022-06-13 at 11 39 01
The URL is correct.

Although, after I run the automated tests, I have some errors (77 times) :

```
Warning: fetchStandardFontData: failed to fetch file "FoxitSans.pfb" with "UnknownErrorException: The standard font "baseUrl" parameter must be specified, ensure that the "standardFontDataUrl" API parameter is provided.".
```

Could you check please ?
Thanks!

**Comment by jf-viguier** [comment]
@florine2623 the error seems related to your Mozilla browser and the pdfjs plugins, not related to my pr. Please check in develop branch.

**Review by florine2623 — APPROVED** [review]
Hello @jf-viguier ,

I've tested again and relaunch the automated tests. I haven't encountered them again.

So for me it is QA ✅ !

**Comment by Progi1984** [comment]
Thanks @jf-viguier & @florine2623

**Comment by Progi1984** [comment]
@PrestaShop/prestashop-maintainers Is this PR needs to be reverted as it brings "BC Breaks"?

**Comment by jf-viguier** [comment]
@Progi1984 I don't think so : the change only check http(s) by default. The previous behaviour was to don't check.

**Pull request merged**